src/modules/roc_pipeline/receiver_session.cpp

```diff
@@ -134,12 +136,11 @@ ReceiverSession::ReceiverSession(const ReceiverSessionConfig& session_config,
             }
             areader = resampler_poisoner_.get();
         }
-
+        audio::SampleSpec sample_spec = audio::SampleSpec(format->sample_spec.getSampleRate(), session_config.sample_spec.getChannels());
         resampler_.reset(audio::ResamplerMap::instance().new_resampler(
                              session_config.resampler_backend, allocator,
                              sample_buffer_pool, session_config.resampler_profile,
-                             common_config.internal_frame_length, format->sample_rate,
-                             session_config.channels),
+                             common_config.internal_frame_length, sample_spec),
                          allocator);
```

Comment:
When I was reviewing ReceiverSession, it was always confusing what is `session_config.sample_spec`.

We have two main sample specs:

* `format->sample_spec` - spec for packets coming from sender
* `common_config.output_sample_spec` - spec for samples coming to mixer, and then passed to audio device

The role of ReceiverSession is to transform the first to the second.

And what is `session_config.sample_spec`? Something intermediate? It's unclear.

This problem is not specific to this PR, it was present before too.

What I suggest is to get rid of `session_config.sample_spec` at all, i.e. remove ReceiverSessionConfig::sample_spec. Every time we need to configure a component with a sample spec, we will use either `format->sample_spec`, `common_config.output_sample_spec`, or create a temporary spec combining rate from one and channels from another.

Can we do it?